### DISCUSSION TITLE
Tests and mock datasets

### DISCUSSION THREAD

**Discussion by BorjaEst** [imk-toolkit/imk-toolkit#67] [Ideas]
Although not all scientists like to use tests, it might be interesting to provide some kind of structure to promote this. If it is easy, people might like it and start using it.

For the structure I do not have any general idea, however I would add the following conditions (if you agree):

- [x] Use of [pytest](https://docs.pytest.org/en/7.0.x/) framework. 
- [ ] Provide a `test data` folder with **small** data files that users could share when creating their tests.

**Comment by ucyo** [comment]
This is definitely an important topic. There is already issue #18 and issue #39 related to this topic open.  The pytest library is anlready included in the dependency list:

https://github.com/imk-toolkit/imk-toolkit/blob/e4b3b597109ed999c2d08d54da6ea27279b49447/pyproject.toml#L33

Please feel free to add a skeleton test framework to run them on the added scripts 🚀 PR 🚀

**Reply by BorjaEst** [reply]
Done at:
https://github.com/imk-toolkit/imk-toolkit/pull/71

Looking forward for comments. Probably it is a bit too complicated but I tried to make it simple for the `user` as I do not expect they will know too much about pytest.

**Reply by ucyo** [reply]
The setup is quite elaborate that is true. Ideally we can provide the scientific user:

1. A template for unit tests 
2. The ability to run these tests with a single command

Everything else should be done on the repo site with as much as possible using CI/CD.

**Reply by BorjaEst** [reply]
Yes, that is what I tried, 1 and 2 are explained on the readme:
 1. __test_template.py
 2. e.g. pytest tests/test_mymethod.py

They should only copy the template and fill the class with their requirements.
(I am open for changes)

**Comment by ucyo** [comment]
Mock data could be randomly generated `xr.Dataset` or `xr.Dataarray` with the possibility to define its seed value for reproducibility.

**Reply by BorjaEst** [reply]
Good point, for now I did just implement a simple "temperature" and "humidity" methods.
However, I can extend them to use seed values.
